src/core/geometry/qgsgeometryeditutils.cpp

```diff
@@ -343,6 +343,10 @@ std::unique_ptr<QgsAbstractGeometry> QgsGeometryEditUtils::avoidIntersections( c
   }
 
   std::unique_ptr< QgsAbstractGeometry > diffGeom( geomEngine->difference( combinedGeometries.get() ) );
+  if ( geomEngine->isEqual( diffGeom.get() ) )
```

Comment:
Can we not doing this test in QgsGeometry::avoidIntersections to avoid changing API of this method (one can expect to have a non-null feature even if there were no modification on the geometry)

Reply:
Sorry, but could you revert this? I didn't see that actually the file is not part of the python API and it's a shame to recreate a geomEngine just to do this operation while we have it in the QgsGeometryEditUtils method.

Could you comment the returned geom in the method documentation to say when a nullptr is returned?